osquery/core/posix/utils.cpp

```diff
@@ -43,4 +43,34 @@ Status platformStrncpy(char* dst, size_t nelms, const char* src, size_t count) {
   ::strncpy(dst, src, count);
   return Status(0, "OK");
 }
+
+const std::string canonicalize_file_name(char* name) {
+  char* buffer = nullptr;
+#ifdef PATH_MAX
+  // On supported platforms where PATH_MAX is defined we can pass null
+  // as buffer, and allow libc to alloced space
+  // On centos/ubuntu libc will use realloc so we will be able to resolve
+  // any path
+  // On darwin libc will allocate PATH_MAX buffer for us
+  char* resolved = realpath(name, nullptr);
+  std::string result = (resolved == nullptr) ? name : resolved;
+  free(resolved);
+#else
+#warning PATH_MAX is undefined, please read comment below
+  // PATH_MAX is not defined, very likely it's not officially supported
+  // os, our best guess is _PC_PATH_MAX if available
+  // In case of failure fallback to "safe" buffer of 8K
+
+  long int path_max = pathconf(name, _PC_PATH_MAX);
+  if (path_max <= 0) {
+    path_max = 8 * 1024;
+  }
+  char* buffer = static_cast<char*>(malloc(path_max));
```

Comment:
Nit. auto

Reply:
IMHO , strict are better than auto, so in case when type is small full type should be used. This is also protects from errors like unexpected auto casts